functional-tests/.sops.yaml

```diff
@@ -1,2 +1,10 @@
 creation_rules:
+    - filename_regex: test_roundtrip_keygroups.yaml
+      key_groups:
+      - pgp: 1022470DE3F0BC54BC6AB62DE05550BC07FB1A0A
+      - pgp: 729D26A79482B5A20DEAD0A76945978B930DD7A2
+    - filename_regex: test_roundtrip_keygroups_missing_decryption_key.yaml
+      key_groups:
+      - pgp: 1022470DE3F0BC54BC6AB62DE05550BC07FB1A0A
+      - pgp: 620B9A4C96230B91E7473D20113D2B26EA0890C7
```

Comment:
per this file structure, are we able to have a mix of pgp and kms keys in a single key group?

Reply:
Going further, since this is a new format, we can avoid repeating the mistakes of the past. I would suggest:

```yaml
creation_rules:
   - filename_regex: *\.yaml$
     key_groups:
        - name: group1
          kms:
            - arn: arn:foo:bar
              role: somerole1
            - arn: arn:foo:baz
              role: somerole2
          pgp:
            - 1022470DE3F0BC54BC6AB62DE05550BC07FB1A0A
            - 620B9A4C96230B91E7473D20113D2B26EA0890C7
        - name: group2
           pgp:
             - 729D26A79482B5A20DEAD0A76945978B930DD7A2
```

Reply:
Yes, a mix of pgp and kms keys is allowed.

We can do what you suggest for the syntax inside kms and pgp entries in each key group, but it might be a bit confusing to have two different syntaxes, one for outside key groups and one for inside.

Reply:
Currently we specify the role inside the ARN. I assume we want to keep doing that?

Reply:
Yeah, from the README:

You can specify a role in the --kms flag and SOPS_KMS_ARN variable by appending it to the ARN of the master key, separated by a + sign:
```
<KMS ARN>+<ROLE ARN>
arn:aws:kms:us-west-2:927034868273:key/fe86dd69-4132-404c-ab86-4269956b4500+arn:aws:iam::927034868273:role/sops-dev-xyz
```

I've made a change so it can be specified like you said. I've also updated the README, and the example in it looks like this:

```
creation_rules:
    - filename_regex: .*keygroups.*
      key_groups:
      # First key group
      - pgp:
        - fingerprint1
        - fingerprint2
        kms:
        - arn: arn1
          role: role1
          context:
            foo: bar
        - arn: arn2
      # Second key group
      - pgp:
        - fingerprint3
        - fingerprint4
        kms:
        - arn: arn3
        - arn: arn4
      # Third key group
      - pgp:
        - fingerprint5
```